components/entity-access-rules/api/src/main/java/org/phenotips/data/permissions/script/PermissionsManagerScriptService.java

```diff
@@ -135,7 +142,28 @@ public PatientAccess getPatientAccess(String targetPatientId)
         if (!this.access.hasAccess(this.userManager.getCurrentUser(), Right.VIEW, patient.getDocumentReference())) {
             return null;
         }
-        return this.manager.getPatientAccess(patient);
+        return this.manager.getEntityAccess(patient);
+    }
+
+    public EntityAccess getEntityAccess(@Nullable final String targetEntityId)
+    {
+                // scripts have only access to a SecurePatient implementation of a Patient,
+        // which does not support all the functionality PatientAccess needs. So
+        // need to get the full Patient object here instead of taking it as an argument
+        //
+        // Since this is a script service, need to check access rights the same way SecurePatientReporistory does.
+        //
+        // TODO: rights management should be refactored so that less is done from velocity
+        //       and this method won't be needed any more
+
+        final PrimaryEntity entity = this.patientRepository.get(targetEntityId);
```

Comment:
Can this be a problem? The input now is not a `patientId`, it is now a `targetEntityId`, so using `patientRepository` to get the object may be a mistake when entity is not a patient. Or if this is only used for patients this class should be renamed.

...as a side note it may be a very difficult task to convert an entity ID to an entity object for a generic unknown Entity type, @itaiGershtansky tried to solve this problem in this entities branch and it is hairy (and may be buggy, still under review)

Reply:
Update: after thinking for a little I think we need to have a central "repository of repositories" (RoR), and whenever we use an entity method with an ID we should also be passing the entity type, which will be used to get the correct repository from that RoR. This can be done as part of Entity refactoring, for now this IS a bug, but since in practice is only used for patients we can let this through if we really need this in this milestone

Reply:
An automatic entity type detection is much more useful in practice than requiring to specify the entity type.

Reply:
It is true, but do we want to do it? This would require figuring out that a "Pxxxxx" is a patient and "FAMxxx" is a family etc., which is doable, but requires a very strong scheme for entity IDs (since security may depend on this)

Reply:
We need to think more from the perspective of those using the API, not those implementing the API. The whole point of the entities architecture was to write less code.

For example, if there's a component implementing a feature for more than one type of entity (in our immediate case, permissions), I shouldn't have to figure out which repository to ask for that entity. I have the entity id, i should get the entity (whatever it is) and process permissions for it. Right now it has to be the usual, ugly "is this a patient? if not, is this a family?"